source/particles/particle_handler.cc

```diff
@@ -1483,7 +1499,440 @@ namespace Particles
+        for (unsigned int i = 0; i < neighbors.size(); ++i)
+          {
+            for (unsigned int j = 0;
+                 j < particles_to_send.at(neighbors[i]).size();
+                 ++j)
+              {
+                particles_to_send.at(neighbors[i])[j]->write_data(data);
+                if (store_callback)
+                  data =
+                    store_callback(particles_to_send.at(neighbors[i])[j], data);
+              }
+          }
+      }
+
+    // Set up the space for the received particle data
+    std::vector<char> recv_data(recv_pointers.back());
+
+    // Exchange the particle data between domains
+    {
+      std::vector<MPI_Request> requests(2 * neighbors.size());
+      unsigned int             send_ops = 0;
+      unsigned int             recv_ops = 0;
+
+      const int mpi_tag = Utilities::MPI::internal::Tags::
+        particle_handler_send_recv_particles_send;
+
+      for (unsigned int i = 0; i < neighbors.size(); ++i)
+        if ((recv_pointers[i + 1] - recv_pointers[i]) > 0)
+          {
+            const int ierr =
+              MPI_Irecv(recv_data.data() + recv_pointers[i],
+                        recv_pointers[i + 1] - recv_pointers[i],
+                        MPI_CHAR,
+                        neighbors[i],
+                        mpi_tag,
+                        parallel_triangulation->get_communicator(),
+                        &(requests[send_ops]));
+            AssertThrowMPI(ierr);
+            send_ops++;
+          }
+
+      for (unsigned int i = 0; i < neighbors.size(); ++i)
+        if ((send_pointers[i + 1] - send_pointers[i]) > 0)
+          {
+            const int ierr =
+              MPI_Isend(send_data.data() + send_pointers[i],
+                        send_pointers[i + 1] - send_pointers[i],
+                        MPI_CHAR,
+                        neighbors[i],
+                        mpi_tag,
+                        parallel_triangulation->get_communicator(),
+                        &(requests[send_ops + recv_ops]));
+            AssertThrowMPI(ierr);
+            recv_ops++;
+          }
+      const int ierr =
+        MPI_Waitall(send_ops + recv_ops, requests.data(), MPI_STATUSES_IGNORE);
+      AssertThrowMPI(ierr);
+    }
+    MPI_Barrier(parallel_triangulation->get_communicator());
+
+    // Put the received particles into the domain if they are in the
+    // triangulation
+    const void *recv_data_it = static_cast<const void *>(recv_data.data());
+
+    // Gather ghost particle iterators from the cache
+    auto &ghost_particles_iterators =
+      ghost_particles_cache.ghost_particles_iterators;
+    unsigned int particle_iterator_counter = 0;
+
+    while (reinterpret_cast<std::size_t>(recv_data_it) -
+             reinterpret_cast<std::size_t>(recv_data.data()) <
+           recv_pointers.back())
+      {
+        auto recv_particle =
+          ghost_particles_iterators[particle_iterator_counter];
+
+        recv_particle->second.free_properties();
```

Comment:
I guess this is indeed needed?

Reply:
Yes! Otherwise the property_pool will crash when it's destructor is called because it still has allocated memory.